lib/view_component/isolated_i18n.rb

```diff
@@ -0,0 +1,44 @@
+require 'set'
+require 'action_view/helpers/translation_helper'
+
+module ViewComponent::IsolatedI18n
+  def self.included(base)
+    base.class_attribute :isolated_i18n_backend, instance_writer: false, instance_predicate: false
+    base.extend ClassMethods
+  end
+
+  Cache = Set.new
+
+  def translate(key = nil, locale: nil, **options) # TODO deprecate :raise
+    locale ||= ::I18n.locale
+    backend = isolated_i18n_backend
+
+    result = catch(:exception) do
+      if key.is_a?(Array)
+        key.map { |k| backend.translate(locale, k, options) }
+      else
+        backend.translate(locale, key, options)
+      end
+    end
+
+    if result.is_a?(I18n::MissingTranslation)
+      handle_exception(false, result, locale, key, options)
+    else
+      result
+    end
```

Comment:
Not really, I18n for some reason (could it be performance?) uses `throw` and `catch` instead of `raise` and `rescue`. The code in this method is copied almost verbatim from I18n, I just removed a things that could be safely cut.